modules/007-registrypackages/images/d8-shutdown-inhibitor/src/pkg/app/tasks/node_cordoner.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2025 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tasks
+
+import (
+	"context"
+	"fmt"
+
+	"d8_shutdown_inhibitor/pkg/kubernetes"
+)
+
+// NodeCordoner waits for shutdown signal and cordons the node.
+type NodeCordoner struct {
+	NodeName         string
+	ShutdownSignalCh <-chan struct{}
+}
+
+func (n *NodeCordoner) Run(ctx context.Context, _ chan error) {
+	// Stage 1. Wait for shutdown.
+	fmt.Printf("nodeCordoner: wait for PrepareForShutdown signal or power key press\n")
+	select {
+	case <-ctx.Done():
+		fmt.Printf("nodeCordoner: stop on global exit\n")
+		return
+	case <-n.ShutdownSignalCh:
+		fmt.Printf("nodeCordoner: catch prepare shutdown signal, cordon node\n")
+	}
+
+	kubectl := kubernetes.NewDefaultKubectl()
+	output, err := kubectl.Cordon(n.NodeName)
```

Comment:
i think we need to check if node is already cordoned

Reply:
if client go has this error

Reply:
It's not an error to issue cordon multiple times:

```
root@static-vm-master-00:~# kubectl cordon static-vm-node-01
node/static-vm-node-01 cordoned
root@static-vm-master-00:~# echo $?
0
root@static-vm-master-00:~# kubectl cordon static-vm-node-01
node/static-vm-node-01 already cordoned
root@static-vm-master-00:~# echo $?
0
```